Source/WebKit/NetworkProcess/cache/NetworkCacheStorage.cpp

```diff
@@ -885,54 +917,63 @@ bool Storage::shouldStoreBodyAsBlob(const Data& bodyData)
 
 void Storage::dispatchWriteOperation(std::unique_ptr<WriteOperation> writeOperationPtr)
 {
-    ASSERT(RunLoop::isMain());
+    assertIsMainThread();
 
     auto& writeOperation = *writeOperationPtr;
-    m_activeWriteOperations.add(WTFMove(writeOperationPtr));
+    auto identifier = writeOperation.identifier();
+    m_activeWriteOperations.add(identifier, WTFMove(writeOperationPtr));
 
+    auto record = writeOperation.record();
     // This was added already when starting the store but filter might have been wiped.
-    addToRecordFilter(writeOperation.record.key);
-
-    backgroundIOQueue().dispatch([this, &writeOperation] {
-        auto recordDirectorPath = recordDirectoryPathForKey(writeOperation.record.key);
-        auto recordPath = recordPathForKey(writeOperation.record.key);
+    addToRecordFilter(record.key);
 
+    backgroundIOQueue().dispatch([this, protectedThis = Ref { *this }, identifier, record = crossThreadCopy(record)]() mutable {
```

Comment:
I agree with Ben, it would be nice to use `crossThreadCopy(WTFMove(record))` here.